android-vjestina-homework-movie-app-data-mock/app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCategoryLabel.kt

```diff
@@ -0,0 +1,59 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import androidx.annotation.StringRes
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+sealed class MovieCategoryLabelTextViewState()
+
+data class MovieCategoryAsString(val value: String) : MovieCategoryLabelTextViewState()
+
+data class MovieCategoryAsStringResource(@StringRes val value: Int) : MovieCategoryLabelTextViewState()
+
+data class MovieCategoryLabelViewState(
+    val itemId: Int,
+    var isSelected: Boolean,
```

Comment:
All ViewStates and their properties MUST be immutable.
When the user clicks a button, the action will be propagated through necessary layers (UI -> Presentation -> Business -> Data) and a new model will be given to the composable function with the state changed (this data will travel from the Data layer in the opposite direction). 
Change `var` to `val`.